roofit/roofitcore/src/RooAbsPdf.cxx

```diff
@@ -1508,9 +1508,10 @@ RooFitResult* RooAbsPdf::fitTo(RooAbsData& data, const RooLinkedList& cmdList)
 	   *obs = *data.get(j);
 	   //Determine first derivatives
 	   std::vector<Double_t> diffs(floated.getSize(), 0.0);
-	   for (int k=0; k<floated.getSize(); k++) {
-	      RooRealVar* paramresult = (RooRealVar*)floated.at(k);
-	      RooRealVar* paraminternal = (RooRealVar*)floatingparams->find(paramresult->getTitle());
+	   for (int k=0; k < floated.getSize(); k++) {
+	      auto paramresult = dynamic_cast<const RooRealVar*>(floated.at(k));
```

Comment:
are you adding const-ness through a dynamic cast? `const auto *paramresult = dynamic/static_cast<RooRealVar*>` might be less surprising